tapchannel/aux_leaf_signer.go

```diff
@@ -735,3 +753,102 @@ func (v *schnorrSigValidator) validateSchnorrSig(virtualTx *wire.MsgTx,
 
 	return nil
 }
+
+// HtlcIndexAsScriptKeyTweak converts the given HTLC index into a modulo N
+// scalar that can be used to tweak the internal key of the HTLC script key on
+// the asset level. The value of 1 is always added to the index to make sure
+// this value is always non-zero.
+func HtlcIndexAsScriptKeyTweak(index input.HtlcIndex) *secp256k1.ModNScalar {
```

Comment:
All four functions below are exported, but aren't used outside of the `tapchannel` package in this PR. Do we expect the need of those functions outside of this package?

Reply:
You're correct. But such stateless and potentially general purpose functions IMO don't hurt to be exported.